components/ExamDetails.tsx

```diff
@@ -26,7 +28,8 @@ export default function ExamDetails({
   isOpen,
   onClose,
 }: ExamDetailsProps) {
-  const { first_name, surname, date_of_birth, sex } = patientData;
+  const [isEditing, setIsEditing] = useState(false);
```

Comment:
I liked the use of this `isEditing` state variable to toggle the form into edit mode.
The only issue I've come across with testing is that if I click edit, and close the dialog (via the close button or Esc), if I click the "View Exam" again, the dialog opens with the form still in edit mode.
I think on close, if the changes weren't saved, it should revert to "view mode"